File: 3st/Jaeyoung/올바른 괄호.js
function solution(s){    
    const stack = [];
    
    for(let i = 0; i < s.length; i++) {
        if (stack[stack.length - 1] === '(' && s[i] === ')') {
            stack.pop();
        } else {
            stack.push(s[i]);
        }
    }
    return !stack.length;
}

File: 3st/Jaeyoung/올바른 괄호.cpp
#include<string>
#include <iostream>
#include <vector>

using namespace std;

bool solution(string s)
{
    bool answer = true;
    vector<int> num;

    for( auto x : s ) {
        if( x == '(' ) {
            num.push_back(x);
        } else {
            if( size( num ) == 0 ) {
                return false;
            }
            num.pop_back();
        }
    }
    
    if( size( num ) == 0 ) {
        return true;
    }

    return false;
}